rc/core/formatter.kak

```diff
@@ -1,13 +1,19 @@
 decl -docstring "shell command to which the contents of the current buffer is piped" \
     str formatcmd
 
-def format -docstring "Format the contents of the current buffer" %{ eval -draft %{
-    %sh{
-        if [ -n "${kak_opt_formatcmd}" ]; then
-            ## Save the current position of the cursor
-            readonly x=$((kak_cursor_column - 1))
-            readonly y="${kak_cursor_line}"
+# This option holds the selection as it was at the time of calling the `format` command
+# After the buffer has been formatted, the selection will be restored according to
+# the value held in this option
+decl -hidden range-specs formatter_prevsel
```

Comment:
I assumed the command would work on any modification made to the buffer, and it seems that this implementation leaves the main selection (and occasionally the secondary selections as well) in the proper location… Should the call to the option be removed, or should the primitives that modify the buffer do a diff?